fdbserver/LogSystem.cpp

```diff
@@ -308,19 +308,19 @@ void LogPushData::writeMessage(StringRef rawMessageWithoutLength, bool usePrevio
 }
 
 void LogPushData::recordEmptyMessage(int loc, const Standalone<StringRef>& value) {
-	if (!isEmptyMessage[loc]) {
+	if (!messagesWritten[loc]) {
 		BinaryWriter w(AssumeVersion(g_network->protocolVersion()));
 		Standalone<StringRef> v = w.toValue();
 		if (value.size() > v.size()) {
-			isEmptyMessage[loc] = true;
```

Comment:
I am fine with this version but my preference is to keep "isEmptyMessage" (and initialize that vector to true) and change this line to "isEmptyMessage[loc] = false" and do "auto count = std::count(isEmptyMessages.begin(), isEmptyMessage.end(), true);" in LogPushData:getEmptyMessageRatio().